_posts/2021-03-04-path-resolution-in-quarkus.adoc

```diff
@@ -0,0 +1,184 @@
+
+As an aside, how we got into this situation is not helped by differences in how libraries behave. Vert.x always wants segments beginning with leading slashes when creating routes, for example, while JAX-RS effectively ignores leading slashes in `@Path` annotations. Anyone used to Vert.x always adds leading slashes, and anyone using JAX-RS just does whatever and it magically works. 
+
+In Quarkus, an implementation detail was exposed by accident: non-application endpoints defined by extensions are based on Vert.x routes. Default path configuration values started with slashes to enable quick route creation and allow simple append behavior. There wasn't anything in the early days of Quarkus to suggest this was a bad idea, and developers with experience in JAX-RS didn't have any warnings one way or the other, because JAX-RS handles it.
+
+Now, however, we were in the situation where paths weren't being resolved as people expected, and the configuration changes required to resolve that situation either weren't intuitive or lead to other problems. We ended up putting all of the possible configuration permutations into a spreadsheet so we could see side by side what happened when you combined different configuration values. The results were not awesome. However, the exercise allowed us to step back and look at the big picture to evaluate what needed to change to allow application and non-application endpoints to behave as you need them to.
+
+While the set of configuration attributes used to configure paths in Quarkus remains unchanged, how configured values are interpreted is different:
+
+* **Endpoint path configuration defaults are now relative values.** `/q` is now `q`, `/metrics` is now `metrics`, etc. This means that, out of the box, these endpoints will resolve relative to the containing root, which is what JAX-RS does per spec, and is what we believe most users intuitively expect.
+* **Leading slashes in explicitly configured values matter.** We know some of you want to move endpoints to specific places, and the most consistent way to express that intent is to allow you to specify the exact uri you want an endpoint to use. If you specify `/metrics`, that is where you will find the metrics endpoint.
+
+These updates have been made available and the dust should all have settled with `1.11.5.Final` and `1.12.1.Final`.
+
+Note that convenience redirects for non-application endpoints are still present, but they can be disabled by setting `quarkus.http.redirect-to-non-application-root-path` to `false`. That hasn't changed at all.
+
+== Resolution of configured paths
+
+Let's go through some examples of how paths resolve using our new rules. We'll start with the following assumptions:
+
+* We have a Hello World application that defines `@ApplicationPath("/hello)`
```

Comment:
```suggestion
* We have a Hello World application that defines `@ApplicationPath("/hello")`
```

Reply:
You're amazing. Thanks for spotting that!